### PR TITLE
Create a cleaning (scrubbing) minigame

### PR DESCRIPTION
**Pull request by josephkimsungha**

### CODE REVIEW

File: src/minigames/scrubMinigame.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+      this.health = Math.max(this.health - event.movement.magnitude() * 0.1, 0);
+      spot.alpha = this.health / 100;
+      if (this.health === 0) {
+        this.delegate.markCleaned(this);
+        spot.removeFromParent();
+      }
+    });
+
+    return spot;
```
Comment (zainafzal08) [inline src/minigames/scrubMinigame.ts:38]: Huh do we need to return spot?

File: src/minigames/scrubMinigame.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+      new DirtySpot(this.app, this),
+    ];
+    dirtySpots.forEach((spot) => {
+      this.remainingDirtySpots.add(spot);
+      this.container.addChild(spot.getObject());
+    });
+
+    this.container.eventMode = "static";
+    this.container.hitArea = this.app.screen;
+    const updateSpotsPointerDown = (isDown: boolean) => {
```
Comment (zainafzal08) [inline src/minigames/scrubMinigame.ts:61]: Is there a reason we don't just have each dirty spot call delegate.isPointerDown and then just have a function on ScrubMinigame that returns of the pointer is currently down?

**Review by zainafzal08 — APPROVED** [review]

File: src/minigames/scrubMinigame.ts (hunk 2 above)
Comment (josephkimsungha) [reply src/minigames/scrubMinigame.ts:61]: I guess we could. But it'll be called for every single mousemove event (of which there'll be a lot)

File: src/minigames/scrubMinigame.ts (hunk 1 above)
Comment (josephkimsungha) [reply src/minigames/scrubMinigame.ts:38]: Yes because the spot is added to the container in line 56

File: src/minigames/scrubMinigame.ts (hunk 2 above)
Comment (zainafzal08) [reply src/minigames/scrubMinigame.ts:61]: It'll be a boolean property lookup we'd set it to true and false based on the pointer down and up events. Basically identical to the current approach just a bit cleaner IMO.

But not fussed either way.

Comment (josephkimsungha) [reply src/minigames/scrubMinigame.ts:61]: Sure, updated in #28

**Pull request merged**